app/src/main/java/com/example/hris/ui/viewmodels/leaves/LeavesViewModel.kt

```diff
@@ -0,0 +1,60 @@
+package com.example.hris.ui.viewmodels.leaves
+
+import android.app.Application
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.viewModelScope
+import com.example.hris.getNumberOfDaysInBetween
+import com.example.hris.model.Leaves
+import com.example.hris.repository.HrisRepository
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class LeavesViewModel @Inject constructor(
+    private val hrisRepository: HrisRepository,
+    application: Application
+) : AndroidViewModel(application) {
+
+    val leaves = MutableLiveData<List<Leaves>>()
+    val loadingDialogState = MutableLiveData<Boolean>()
+    val message = MutableLiveData<String>()
+    val totalVacationLeavesLeft = MutableLiveData<Double>()
+    val totalSickLeavesLeft = MutableLiveData<Double>()
```

Comment:
1. Make this a `MediatorLiveData` that uses `leaves` as its source.
2. Use `.filter {}` and `.sumOf {}`
3. Move the computation of the number of days in between inside the `Leave` class.